### ISSUE TITLE
Azure Error: MissingApiVersionParameter  (securitycenter.py)

### ISSUE THREAD

**Issue by ab-smith** [nccgroup/ScoutSuite#1385]
**Describe the bug**

Steps:

- downloaded the package using pip (or master branch on git)
- follow the basic steps to start the audit (using account browser since I have MFA)
- Get this error and scout is just stuck:

ERROR securitycenter.py L86: Failed to retrieve compliance results: Azure Error: MissingApiVersionParameter
Message: The api-version query parameter (?api-version=) is required for all requests.

**To Reproduce**

Please provide:

- python3 scout.py azure --user-account-browser --tenant xxx-yyy-zzzz

**Additional context**

Same behaviour using git version or pip version

**Comment by ab-smith** [comment]
Sorry if there is a duplicate that I haven't found.

**Comment by bartosz-piotrowski-a** [comment]
UP: I have same issue.

**Comment by stephenc01** [comment]
Any quick work around for this?

**Comment by nes-jisc** [comment]
Same issue here

**Comment by lshw54** [comment]
Same issue

**Comment by mhomrane** [comment]
Getting the same issue too.

**Comment by duckie37** [comment]
Same issue..

**Comment by duckie37** [comment]
I got around this error  by downgrading to  azure-cli=2.5.1-1~bionic .  I did see the error pop up again in some sections and since this is my first time using this tool I am not sure what it typically looks like.

**Comment by AbhishekDptr** [comment]
Same issue here. I am stuck with this now. Please provide some workaround atleast for the time being

**Comment by DjangoDani** [comment]
Any news regarding this issue? The report of security center would be really usefull

**Comment by maartenvdezz** [comment]
Same issue here

**Comment by billabongrob** [comment]
Same issue here.

**Comment by LexStar** [comment]
still same issue

**Comment by arunksingh16** [comment]
I am also facing this issue. 

2022-04-21 10:42:40 XXXXXXX scout[5043] INFO Fetching resources for the App Services service
2022-04-21 10:42:41 XXXXXXX scout[5043] ERROR securitycenter.py L86: Failed to retrieve compliance results: Azure Error: MissingApiVersionParameter
Message: The api-version query parameter (?api-version=) is required for all requests.

**Comment by vollkorn1982** [comment]
I am affected as well:

2022-04-26 19:54:23 jan-ThinkPad-T14s scout[42266] ERROR securitycenter.py L86: Failed to retrieve compliance results: Azure Error: MissingApiVersionParameter
Message: The api-version query parameter (?api-version=) is required for all requests.

Running on Ubuntu 20.04, fresh git clone from today in a venv with Python 3.8.10

**Comment by jurajsucik** [comment]
Same issue here

**Comment by phillips321** [comment]
Same issue here

**Comment by bzogala** [comment]
Still present in 5.11. Running on Mac with homebrew python and seperate venv dedicated for scout. Besides patch do we have conclusion what is core issue here ? Is the error within azure-cli or scout itself?  Shouldn't scout handle it better ?

**Comment by stephenc01** [comment]
I believe azure their endpoint and requires to send the api version in the request. IMO scout needs to have this requirement incorporated.

Thank You,
Stephen C
-All of my email addresses go to the same place
-Save Paper, think before you print

**Comment by bzogala** [comment]
This is message coming from console, but doesn't give any specific information on actual implementation. And it means something is inconsistent either on scout side or azure library or azure backend. As the error doesn't appear globally, only in security center

**Comment by Popzi** [comment]
@fernando-gallego looks like the issue will be resolved in the **develop** branch, but also after the merge of #1425 Graph API will only return 100 results and Microsoft refers to this as Paging:  https://docs.microsoft.com/en-us/graph/paging

**Comment by michyweb** [comment]
@Popzi could you please create an issue for this?

**Comment by fernando-gallego** [comment]
Workaround implemented in version 5.12.0, following up in https://github.com/nccgroup/ScoutSuite/issues/1428

**Comment by tkmru** [comment]
I'm getting `MissingApiVersionParameter` error even with v5.12.0.

```
scout[15213] ERROR securitycenter.py L86: Failed to retrieve compliance results: Azure Error: MissingApiVersionParameter
Message: The api-version query parameter (?api-version=) is required for all requests.
```
